server/app/com/xsn/explorer/util/RetryableFuture.scala

```diff
@@ -0,0 +1,59 @@
+package com.xsn.explorer.util
+
+import scala.concurrent.{ExecutionContext, Future}
+import akka.actor.Scheduler
+import akka.pattern.Patterns.after
+
+import scala.concurrent.duration._
+import scala.util.Try
+
+object RetryableFuture {
+
+  def apply[A](
+      delays: List[FiniteDuration]
+  )(
+      shouldRetry: Try[A] => Boolean
+  )(f: => Future[A])(implicit ec: ExecutionContext, scheduler: Scheduler): Future[A] = {
+    delays.foldLeft(f) {
+      case (result, delay) =>
+        result.transformWith { t =>
+          if (shouldRetry(t)) {
+            after(delay, scheduler, ec, Future.successful(1)).flatMap(_ => f)
```

Comment:
Another way is `scheduler.scheduleOnce(delay)(f)`, it looks simpler to me.

Reply:
`scheduler.scheduleOnce` returns Cancellable while `after` returns a future so i could not make this change

Reply:
Oh, nice catch.